normandy/selfrepair/static/js/self_repair_runner.js

```diff
@@ -103,14 +100,46 @@ function runRecipe(recipe) {
 }
 
 
+/**
+ * Generate a context object for JEXL filter expressions.
+ * @return {object}
+ */
+function filterContext() {
+    return {
+        normandy: classifyClient(),
+        telemetry: {},
+        events: {},
+    }
+}
+
+
+/**
+ * Match a recipe against a context using its filter expression.
+ * @param  {Recipe} recipe  Recipe fetched from the server.
+ * @param  {object} context Context returned by filterContext.
+ * @promise Resolves with a list containing the recipe and a boolean
+ *     signifying if the filter passed or failed.
+ */
+function matches(recipe, context) {
+    return jexl.eval(recipe.filter_expression, context)
+    .then(value => [recipe, !!value]);
```

Comment:
If we're going with "empty matches everything", you'll need some handling of that here.

Also, Jexl doesn't do multiline expression, and we have the habit of pressing Enter while writing filters. This should join the `filter_expression` into one line.